### PR TITLE
Add resolve_local_name to resolve names in a function scope

### PR DESCRIPTION
**Pull request by kjeremy**
First step to resolving #80

### CODE REVIEW

File: crates/ra_editor/src/scope/fn_scope.rs [hunk 1]
```diff
@@ -241,6 +241,18 @@ struct ScopeData {
     entries: Vec<ScopeEntry>
 }
 
+pub fn resolve_local_name<'a>(name_ref: ast::NameRef, scopes: &'a FnScopes) -> Option<ast::Name<'a>> {
```
Comment (matklad) [inline crates/ra_editor/src/scope/fn_scope.rs]: I think you can return a `ScopeEntry` here, and convert it to a `Name` on the call site.

Comment (kjeremy) [reply crates/ra_editor/src/scope/fn_scope.rs]: If I return `Option<&'a ScopeEntry>` I can get at the name as a `SmolStr`. I can't get the `Name` without making `ScopeEntry::ast()` public

Comment (matklad) [reply crates/ra_editor/src/scope/fn_scope.rs]: I think it's OK making that public!

Comment (kjeremy) [reply crates/ra_editor/src/scope/fn_scope.rs]: Done.

File: crates/ra_editor/src/scope/fn_scope.rs [hunk 2]
```diff
@@ -326,4 +337,69 @@ mod tests {
...
+        );
+    }
+
+    fn do_check_local_name(code: &str, expected_offset: u32) {
+        let (off, code) = extract_offset(code);
+        let code = {
+            let mut buf = String::new();
+            let off = u32::from(off) as usize;
+            buf.push_str(&code[..off]);
+            buf.push_str(&code[off..]);
```
[the remark this answers is not in the document]
Comment (kjeremy) [reply crates/ra_editor/src/scope/fn_scope.rs]: Good catch.

**Comment by kjeremy** [comment]
bors: r+

**Pull request merged**